plutus/budget_manager/gcp_helper.py

```diff
@@ -0,0 +1,283 @@
+from constants import APP
+import copy
+import logging
+import markus
+from protobuf_to_dict import protobuf_to_dict
+import sys
+
+metrics = markus.get_metrics(APP + ".gcphelper")
+
+class GcpHelper:
+    '''
+    This class uses GCP billing budget and resource manager clients to get, update, and create budgets
+    '''
+
+    def __init__(self, billing_client, rm_client):
+        self.billing_client = billing_client
+        self.rm_client = rm_client
+        self.logger = logging.getLogger(APP + ".gcphelper")
+
+
+    def get_project_number(self, project_id):
+        ''' Given a GCP project id, return the associated GCP project number. '''
+        # Ensure that the configured project_id actually exists in GCP
+        try:
+            project_object = self.rm_client.fetch_project(project_id)
+            metrics.incr("gcp_api_request_count", tags = ["type:resource_manager.fetch", f"project_id:{project_id}"])
+            project_number = project_object.number
+            return project_number
+        except Exception as err:
+            self.logger.error(f"Request to fetch project id {project_id} failed due to: {err}. Please double check your configuration.")
+            metrics.incr("error_count", tags = ["type:misconfig", f"project_id:{project_id}"])
```

Comment:
Is tag `type:misconfig` be correct here? An exception can occur here without being a misconfiguration.

Reply:
thx